### ISSUE TITLE
cray-mpich toolchain not recognized by MPIPreferences

### ISSUE THREAD

**Issue by williamfgc** [JuliaParallel/MPI.jl#613]
@luraess and I have been testing MPI.jl on OLCF's Crusher which uses cray-mpich. 
Runs execute as independent MPI processes, we think the issue is that MPIPreferences can't recognize the cray-mpich system MPI. To reproduce: 

```
module load rocm cray-mpich

julia> using MPIPreferences

julia> MPIPreferences.use_system_binary()
ERROR: MPI library could not be found
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] use_system_binary(; library_names::Vector{String}, mpiexec::String, abi::Nothing, export_prefs::Bool, force::Bool)
   @ MPIPreferences ~/.julia/packages/MPIPreferences/MXVwb/src/MPIPreferences.jl:130
 [3] use_system_binary()
   @ MPIPreferences ~/.julia/packages/MPIPreferences/MXVwb/src/MPIPreferences.jl:117
 [4] top-level scope
   @ REPL[2]:1
```

cray-mpich doesn't have a `libmpi.so` but `libmpi_cray.so` and uses `srun` as the executable:

```
       libmpi_cray.so.12 (libc6,x86-64) => /opt/cray/pe/lib64/libmpi_cray.so.12
	libmpi_cray.so (libc6,x86-64) => /opt/cray/pe/lib64/libmpi_cray.so
	libmpi_amd.so.12 (libc6,x86-64) => /opt/cray/pe/lib64/libmpi_amd.so.12
	libmpi_amd.so (libc6,x86-64) => /opt/cray/pe/lib64/libmpi_amd.so
```

Just looking for guidance or a potential feature support. We'll be happy to help make this functional and test on Crusher.

**Comment by giordano** [comment]
You can use `MPIPreferences.use_system_binary(; library_names=["libmpi_cray.so"], mpiexec="srun")`: https://juliaparallel.org/MPI.jl/dev/reference/mpipreferences/#MPIPreferences.use_system_binary

**Comment by williamfgc** [comment]
> You can use

MPIPreferences.use_system_binary(; library_names=["libmpi_cray"], mpiexec="srun")

@giordano thanks, this made `hello world` work and recognize the 8 GPUs on one node. There are errors in the miniapp, complaining about requesting more devices and a seg fault on a different test. We are currently debugging, but this helped already to move forward. We'll keep you posted.

**Comment by luraess** [comment]
I am testing Julia MPI on CSC LUMI using `#master` branch of MPI, and still had to execute `MPIPreferences.use_system_binary(; library_names=["libmpi_cray"], mpiexec="srun")` as https://github.com/JuliaParallel/MPI.jl/pull/614 seems not to work yet. Any idea what could still make it not discoverable @giordano ?

**Comment by giordano** [comment]
> I am testing Julia MPI on CSC LUMI using `#master` branch of MPI, and still had to execute `MPIPreferences.use_system_binary(; library_names=["libmpi_cray"], mpiexec="srun")` as https://github.com/JuliaParallel/MPI.jl/pull/614 seems not to work yet. Any idea what could still make it not discoverable @giordano ?

I just ran into this myself on a Cray system.  Having to specify `mpiexec="srun"` is expected, see https://github.com/JuliaParallel/MPI.jl/pull/614#issuecomment-1161917327, but I don't quite understand why I also had to pass `library_names=["libmpi_cray"]`.  I'll see if I can find some time to debug this.

**Comment by giordano** [comment]
I'm going to close this ticket as I believe the original issue was fixed by #614 (but remember you may need to change `mpiexec` value if you want to use `srun`).  For any other problems with Cray MPICH, please do open a new issue!
